File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i = 0; i < s.length(); i++){
            if((stack.size()>0)&&(stack.peek()=='('&&s.charAt(i)==')'||stack.peek()=='['&&s.charAt(i)==']'||stack.peek()=='{'&&s.charAt(i)=='}')){
                stack.pop();
            } else {
                stack.push(s.charAt(i));
            }
        }
        return stack.size()==0;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        l = []
        for i in range(0, len(s)):
            ch = s[i]
            if (ch==')' and len(l) > 0 and l[-1]=='('):
                l.pop()
            elif (ch=='}' and len(l) > 0 and l[-1]=='{'):
                l.pop()
            elif (ch==']' and len(l) > 0 and l[-1]=='['):
                l.pop()
            else :
                l.append(ch)
        
        if len(l)==0:
            return True
        else:
            return False